.github/actions/restore-node/check-git-status.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# Navigate to the specified directory
+cd $1
+# Set verbose execution
+set -x
+# Check for unexpected changes
+# Fail if git status detects changes, ignoring:
+# - Files ending with 'junit.xml'
+# - The untracked file 'endo-sha.txt'
+# This setup ensures the Git status only flags unexpected modifications or untracked files outside these specified exceptions.
+changes=$(git status --porcelain | grep -vE 'junit.xml$' | grep -vE '^\?\? endo-sha.txt$')
```

Comment:
This should be maximally strict, and if it is using separate `grep` invocations, it might as well improve readability by avoiding regular expressions where possible:
```suggestion
changes=$(git status --porcelain | grep -vE '/junit\.xml$' | grep -vFx '?? endo-sha.txt')
```